test/unit_tests/braket/default_simulator/test_noise_operations.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2019-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import pytest
+
+import braket.ir.jaqcd as instruction
+from braket.default_simulator import noise_operations
+from braket.default_simulator.operation_helpers import check_cptp, from_braket_instruction
+from braket.ir.jaqcd import shared_models
+
+testdata = [
```

Comment:
Can you add some test cases with probability = 0? with probability = 1? with probability = float("nan")? What happens with invalid Kraus matrices (e.g. 3x3)? What happens when targets size doesn't match the matrix size for Kraus matrices?

Reply:
Do we expect probability=1 work? Now it fails for probability=1 but works for probability=1.0

Reply:
Also, @avawang1 @kshitijc , do we want to add checks in local simulator repo? I saw GateOperation have no checks for angle and target, so I didn't add checks for KrausOperation neither. All checks are made in SDK and IR

Reply:
There may be a (not likely) scenario where the user does not use the SDK and just passes in a not-validated ir using the `construct` function in Pydantic. If you have time, feel free to add checks in this repo as well.

Reply:
@avawang1 To add checks to angle and probability, we might need to add classes for AngleGate and ProbabilisticNoise like we do in SDK. But I think it's would be good that we don't add more codes to the pull request because it's already big. So can we add checks later after this pull request?

Reply:
Sure, np